Code/Framework/AzFramework/AzFramework/Network/IRemoteTools.h

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/base.h>
+#include <AzCore/EBus/Event.h>
+#include <AzCore/Memory/OSAllocator.h>
+#include <AzCore/Memory/SystemAllocator.h>
+#include <AzCore/RTTI/ReflectContext.h>
+#include <AzCore/RTTI/RTTI.h>
+#include <AzCore/Serialization/SerializeContext.h>
+#include <AzCore/std/containers/deque.h>
+#include <AzCore/std/containers/unordered_map.h>
+#include <AzCore/std/smart_ptr/intrusive_ptr.h>
+#include <AzCore/std/smart_ptr/intrusive_refcount.h>
+#include <AzCore/std/string/string.h>
+
+namespace AZ
+{
+    template<typename T>
+    class Interface;
+
+    class Name;
+}
+
+namespace AzFramework
+{
+    class RemoteToolsMessage : public AZStd::intrusive_refcount<size_t>
+    {
+    public:
+        AZ_CLASS_ALLOCATOR(RemoteToolsMessage, AZ::OSAllocator, 0);
+        AZ_RTTI(RemoteToolsMessage, "{8512328C-949D-4F0C-B48D-77C26C207443}");
+
+        RemoteToolsMessage()
+            : m_msgId(0)
```

Comment:
Move these initializers to the member declarations themselves, then these constructors can be most just default.

```
        AZ::u64 m_msgId = 0;
        AZ::u32 m_senderTargetId = 0;
        const void* m_customBlob = nullptr;
        AZ::u32 m_customBlobSize = 0;
etc...
```

and then
```
RemoteToolsMessage() = default;
RemoteToolsMessage(AZ::u64 msgId) : m_msgId(msgId) {}
```

Reply:
Also can you make the `RemoteToolsMessage(AZ::u64 msgId)` explicit as well?